### PR TITLE
Table with Filters

### PR DESCRIPTION
**Pull request by paladin-tech**
### Description
EuiSuggest / OUI implementation of Table with Filters UI

### Issues Resolved
https://github.com/opensearch-project/security-analytics-dashboards-plugin/issues/13
https://github.com/opensearch-project/security-analytics-dashboards-plugin/issues/14

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/TableWithFilters/GlobalFilterOptions.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+  if (tree.items) {
+    tree.items.forEach((item: any) => {
+      if (item.panel) {
+        flattenPanelTree(item.panel, array);
+        item.panel = item.panel.id;
+      }
+    });
+  }
+
```
Comment (phestecAMZN) [inline public/components/TableWithFilters/GlobalFilterOptions.tsx:27]: Second "flattenPanelTree"

File: public/components/TableWithFilters/GlobalFilterItem.tsx [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+
+  if (tree.items) {
+    tree.items.forEach((item) => {
+      if (item.panel) {
+        flattenPanelTree(item.panel, array);
+        item.panel = item.panel.id;
+      }
+    });
+  }
+
```
Comment (phestecAMZN) [inline public/components/TableWithFilters/GlobalFilterItem.tsx:30]: Maybe move this out into a helpers file and share with GlobalFilterOptions.tsx?

**Review by phestecAMZN — APPROVED** [review]
just the one item, not critical

**Comment by amsiglan** [comment]
Will create another PR with latest changes

**Pull request closed without merging**